pkg/load/agents/utils.go

```diff
@@ -62,3 +63,72 @@ func startWatchers(path string, callback func() error, recordError func(string))
 	}
 	return nil
 }
+
+func startReleaseRepoWatcher(path string, callback func() error, recordError func(string)) error {
+	errFunc := func(err error, msg string) {
+		recordError(msg)
+		logrus.WithError(err).Error(msg)
+	}
+
+	watcher, err := GetSymlinkWatcher(callback, errFunc, path)
+	if err != nil {
+		return fmt.Errorf("failed to get the symblink watcher: %w", err)
+	}
+	interrupts.Run(watcher)
+
+	return nil
+}
+
+// GetSymlinkWatcher returns a function that watches if the destination of a single symbolic link changed and run
+// the event function against the new destination.
+func GetSymlinkWatcher(eventFunc func() error, errFunc func(error, string), path string) (func(ctx context.Context), error) {
+	w, err := fsnotify.NewWatcher()
+	if err != nil {
+		return nil, err
+	}
+
+	oldSymlink, err := os.Readlink(path)
```

Comment:
There is a race here.  The link can change between this and the `Add` below.

Reply:
This could happen:

- `oldSymlink = "d0"`
- `git-sync` updates the directory and removes `d0`
- `w.Add` below will try to add `d0` and fail

Reply:
`w.Add` will add the new symlink instead.

Reply:
It will not, `oldSymlink` will not have changed.

Reply:
At this point, we just initialize the current symlink. By the time we will add the new folder to the watcher, the `oldSymlink` will change.